Highway.cpp

```diff
@@ -13,34 +15,54 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    if(Motorcycle* m = dynamic_cast<Motorcycle*>(v))
+    {
+        m->setSpeed(speedLimit);
+    }
+    if(Car* c = dynamic_cast<Car*>(v))
+    {
+        c->setSpeed(speedLimit);
+    }
+    if(SemiTruck* t = dynamic_cast<SemiTruck*>(v))
+    {
+        t->setSpeed(speedLimit);
+    }
```

Comment:
use `else if`
you're still checking if it is a Car, even if it's a motorcycle
Also, use `auto* m = dynamic_cast...` so you don't write the Type name twice
- [ ] revise